File: implementBST.java
import java.util.*;
class TreeNode{
  int val;
  TreeNode left;
  TreeNode right;

  TreeNode(int x) {
    val = x;
  }
}

class BST{

  TreeNode root;

  // Iterative Approach TC O(log n) SC O(1)
  public void insertIntoBST(int key){
    TreeNode node=new TreeNode(key);
    if(root==null) {
        root = node;
        return;
    }
    TreeNode prev=null;
    TreeNode temp=root;
    while (temp!=null){
        if(temp.val>key){
            prev=temp;
            temp=temp.left;
        }
        else if (temp.val<key){
            prev=temp;
            temp=temp.right;
        }
    }
    if(prev.val>key)
        prev.left=node;
    else prev.right=node;
  }

  public void inorder(){
    TreeNode temp=root;
    Stack<TreeNode> stack=new Stack<>();
    while (temp!=null||!stack.isEmpty()){
        if(temp!=null){
            stack.add(temp);
            temp=temp.left;
        }
        else {
            temp=stack.pop();
            System.out.print(temp.val+" ");
            temp=temp.right;
        }
      }
  }
}

class implementBST{
  public static void main(String [] args){
    BST bst= new BST();
    bst.insertIntoBST(8);
    bst.insertIntoBST(3);
    bst.insertIntoBST(7);
    bst.insertIntoBST(5);
    bst.insertIntoBST(1);
    bst.inorder();
  }
}

File: implementBST.py
class TreeNode:
	def __init__(self, val=0, left=None, right=None):
		self.val=val 
		self.left=left
		self.right=right


class BST:
	def __init__(self):
		self.root=TreeNode()

	def insert(self, root, val):
		if not root:
			return TreeNode(val)

		if val<root.val:
			root.left=self.insert(root.left, val) 
		else:
			root.right=self.insert(root.right, val)

		return root 

	def constructBST(self, keys):
		self.root=None  
		for key in keys:
			self.root=self.insert(self.root, key)

		return self.root

	def inorder(self, root):
		if not root: return 

		self.inorder(root.left)
		print(root.val, end=" ")
		self.inorder(root.right)

# bst=BST()
# keys=[4,2,6,1,3]
# root=bst.constructBST(keys)
# bst.inorder(root) 